docker-compose.yml

```
@@ -0,0 +1,150 @@
+version: '3.7'
+services:
+  provisioning-db:
+    restart: on-failure
+    image: docker.io/postgres:10.5
+    healthcheck:
+      test: [ "CMD", "pg_isready", "-U", "postgres", "-d", "postgres" ]
+      interval: 2s
+      timeout: 3s
+      retries: 10
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
+      - POSTGRES_DB=provisioning
+    volumes:
+      - ./postgres-data:/var/lib/postgresql/data
+    ports:
+      - 5432:5432
+    
+  redis:
+    image: docker.io/redis:4.0.11
+    healthcheck:
+      test: [ "CMD", "redis-cli", "ping" ]
+      interval: 2s
+      timeout: 2s
+      retries: 10
+    ports:
+      - 6379:6379
+  zookeeper:
+    profiles:
+      - "kafka"
+    image: confluentinc/cp-zookeeper
+    environment:
+      - ZOOKEEPER_CLIENT_PORT=32181
+      - ZOOKEEPER_SERVER_ID=1
+    expose:
+      - 32181
+  kafka:
+    profiles:
+      - "kafka"
+    image: confluentinc/cp-kafka
+    environment:
+      - KAFKA_ADVERTISED_LISTENERS=DOCKER://kafka:29092,LOCALHOST://localhost:9092
+      - KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=DOCKER:PLAINTEXT,LOCALHOST:PLAINTEXT
+      - KAFKA_INTER_BROKER_LISTENER_NAME=DOCKER
+      - KAFKA_BROKER_ID=1
+      - KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1
+      - KAFKA_AUTO_CREATE_TOPICS_ENABLE=true
+      - KAFKA_ZOOKEEPER_CONNECT=zookeeper:32181
+    ports:
+      - 9092:9092
+      - 29092:29092
+    depends_on:
+      - zookeeper
+
+  init-kafka:
+    profiles:
+      - "kafka"
+    image: confluentinc/cp-kafka
+    entrypoint: [ '/bin/sh', '-c' ]
+    command: |
+      "
+      # blocks until kafka is reachable
+      kafka-topics --bootstrap-server kafka:29092 --list
+
+      echo -e 'Creating kafka topics'
+      kafka-topics --bootstrap-server kafka:29092 --create --if-not-exists --topic platform.sources.status --replication-factor 1 --partitions 1
+      kafka-topics --bootstrap-server kafka:29092 --create --if-not-exists --topic platform.provisioning.internal.availability-check --replication-factor 1 --partitions 1
+
+      echo -e 'Successfully created the following topics:'
+      kafka-topics --bootstrap-server kafka:29092 --list
+      "
+    depends_on:
+      - kafka
+      - zookeeper    
+  backend:
+    env_file:
+      - config/api.env
+    depends_on:
+      migrate:
+        condition: service_completed_successfully
+      provisioning-db:
+        condition: service_healthy
+    build:
+      context: .
+      dockerfile: build/Dockerfile.dev
+    ports:
+      - 8000:8000
+    volumes:
+      - go-modules:/go/pkg/mod # Put modules cache into a separate volume
+
+  migrate: # This service is used to run migrations
+    profiles:
+      - "migrate"
+    build:
+      context: .
+      dockerfile: build/Dockerfile.dev
+    env_file:
+      - config/api.env
+    depends_on:
+      provisioning-db:
+        condition: service_healthy
+    entrypoint: [/pbmigrate]
```

```suggestion
    entrypoint: ["go", "run", "/build/pbmigrate]
```

if you would stick to my suggestion to use Go run instead compilation.